### PR TITLE
JavaDocs for Item and Item.Settings

### PR DESCRIPTION
**Pull request by Draylar**
This WIP PR adds in a few new JavaDocs for `Item` and `Item.Settings`, partially based on the [overview I wrote on the wiki](https://fabricmc.net/wiki/tutorial:items_docs) a while back.

I do have a few questions on what I should mention. From the context of an Item method, would it be good to point out how the method is only used in ItemStacks (eg. _"Defines the maximum durability of an ItemStack with this Item"_ vs. _"Defines the maximum durability of this Item"_)? Similarly, should `Item.Settings` docs mention the `Item` they are attached to (_"Sets the maximum stack count of any Item using this Item.Settings instance"_ vs. _"Sets the maximum stack count of an Item"_ vs. _"Sets the maximum stack count of this Settings instance"_)?

Looking forward to any feedback. Thanks in advance!

### CODE REVIEW

File: mappings/net/minecraft/item/Item.mapping [hunk 1]
```diff
@@ -59,6 +63,7 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 	METHOD method_7844 getAttributeModifiers (Lnet/minecraft/class_1304;)Lcom/google/common/collect/Multimap;
 		ARG 1 slot
 	METHOD method_7846 isDamageable ()Z
+		COMMENT @return whether an ItemStack with this Item can loose durability.
```
Comment (Gegy) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT @return whether this Item can lose durability.
```
The mention of `ItemStack` here is unique from other similar docs, so might be worth excluding it here

File: mappings/net/minecraft/item/Item.mapping [hunk 2]
```diff
@@ -97,6 +102,12 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		ARG 2 world
 		ARG 3 user
 	METHOD method_7862 getRarity (Lnet/minecraft/class_1799;)Lnet/minecraft/class_1814;
+		COMMENT Returns this Items's {@link Rarity}, which changes the color of its name.
```
Comment (Gegy) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT Returns this Item's {@link Rarity}, which changes the color of its name.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]
Make sure to use lowercase `item` versus `Item` in sentences.

**Comment by Draylar** [comment]
> Make sure to use lowercase `item` versus `Item` in sentences.

The intention was to point to the Item class without linking to it. Does it make more sense to keep it lowercase?

**Review by jaskarth — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

File: mappings/net/minecraft/item/Item.mapping [hunk 3]
```diff
@@ -214,17 +231,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		FIELD field_8019 maxDamage I
 		FIELD field_8020 maxCount I
 		METHOD method_19265 food (Lnet/minecraft/class_4174;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT When set, any item configured with this Settings instance will be edible based on the provided {@link FoodComponent}.
+			COMMENT
+			COMMENT @param foodComponent configured food properties for any item using this Settings instance
```
Comment (YanisBft) [inline mappings/net/minecraft/item/Item.mapping]: Please put the docs in the param itself (same for all the builder methods)

File: mappings/net/minecraft/item/Item.mapping [hunk 4]
```diff
@@ -18,11 +18,14 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 	METHOD method_19263 isFood ()Z
 		COMMENT Checks if this item is food and therefore is edible.
 	METHOD method_19264 getFoodComponent ()Lnet/minecraft/class_4174;
+		COMMENT @return this item's {@link FoodComponent}, or null if one was not set.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT Returns this item's {@link FoodComponent}, or {@code null} if one was not set.
```
Like in google java style, we don't use return tag if the return info is the only statement in the method description. When we are using java 15 to build, we can use `{@return}` inline tag at that time.

File: mappings/net/minecraft/item/Item.mapping [hunk 5]
```diff
@@ -18,11 +18,14 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
 	METHOD method_21830 getEatSound ()Lnet/minecraft/class_3414;
 	METHOD method_21831 getDrinkSound ()Lnet/minecraft/class_3414;
 	METHOD method_24357 damage (Lnet/minecraft/class_1282;)Z
+		COMMENT @return whether this item can be damaged by the given {@link DamageSource}.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT Returns whether this item can be damaged by the given {@link DamageSource source}.
```
This renders like "Returns whether this item can be damaged by the given `source`"

File: mappings/net/minecraft/item/Item.mapping [hunk 6]
```diff
@@ -18,11 +18,14 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
 		ARG 1 source
 	METHOD method_24358 isFireproof ()Z
+			COMMENT @return whether this item is immune to fire and lava damage.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
			COMMENT Returns whether this item is immune to fire and lava damage.
```

File: mappings/net/minecraft/item/Item.mapping [hunk 7]
```diff
@@ -77,6 +80,7 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		ARG 3 user
 		ARG 4 remainingUseTicks
 	METHOD method_7841 getMaxDamage ()I
+		COMMENT @return the maximum durability of this item. Can be configured through {@link Item.Settings}.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT Returns the maximum durability of this item. Can be configured through {@link Item.Settings#maxDamage(int) settings.maxDamage()}.
		COMMENT
		COMMENT @return the maximum durability of this item
```

File: mappings/net/minecraft/item/Item.mapping [hunk 8]
```diff
@@ -85,6 +89,7 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 	METHOD method_7844 getAttributeModifiers (Lnet/minecraft/class_1304;)Lcom/google/common/collect/Multimap;
 		ARG 1 slot
 	METHOD method_7846 isDamageable ()Z
+		COMMENT @return whether this item can lose durability.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT Returns whether this item can lose durability.
```

File: mappings/net/minecraft/item/Item.mapping [hunk 9]
```diff
@@ -128,6 +133,12 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		ARG 2 world
 		ARG 3 user
 	METHOD method_7862 getRarity (Lnet/minecraft/class_1799;)Lnet/minecraft/class_1814;
+		COMMENT Returns this item's {@link Rarity}, which changes the color of its name.
+		COMMENT
+		COMMENT <p>By default, if an item has an enchantment, its rarity is modified:
+		COMMENT <ul>
+		COMMENT \t<li>Common and Uncommon -> Rare
+		COMMENT \t<li>Rare -> Epic
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:141]:
```suggestion
		COMMENT \t<li>Rare -> Epic
		COMMENT </ul>
```

File: mappings/net/minecraft/item/Item.mapping [hunk 10]
```diff
@@ -172,8 +186,11 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 	METHOD method_7880 getRawId (Lnet/minecraft/class_1792;)I
 		ARG 0 item
 	METHOD method_7881 getMaxUseTime (Lnet/minecraft/class_1799;)I
+		COMMENT Returns the maximum use (right-click) time of this item, in ticks.
+		COMMENT Once a player has used an item for said number of ticks, they stop using it, and {@link Item#finishUsing} is called.
 		ARG 1 stack
 	METHOD method_7882 getMaxCount ()I
+		COMMENT @return the maximum stack count of any ItemStack with this item. Can be configured through {@link Item.Settings}.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
		COMMENT @return the maximum stack count of any ItemStack with this item. Can be configured through {@link Item.Settings#maxCount(int) settings.maxCount()}.
```

File: mappings/net/minecraft/item/Item.mapping [hunk 11]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		FIELD field_8019 maxDamage I
 		FIELD field_8020 maxCount I
 		METHOD method_19265 food (Lnet/minecraft/class_4174;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT When set, any item configured with this Settings instance will be edible based on the provided {@link FoodComponent}.
+			COMMENT
+			COMMENT @param foodComponent configured food properties for any item using this Settings instance
+			COMMENT @return this instance
 			ARG 1 foodComponent
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:237]:
```suggestion
			ARG 1 foodComponent
				COMMENT configured food properties for any item using this Settings instance
```

File: mappings/net/minecraft/item/Item.mapping [hunk 12]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
 		FIELD field_8019 maxDamage I
 		FIELD field_8020 maxCount I
 		METHOD method_19265 food (Lnet/minecraft/class_4174;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT When set, any item configured with this Settings instance will be edible based on the provided {@link FoodComponent}.
+			COMMENT
+			COMMENT @param foodComponent configured food properties for any item using this Settings instance
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 13]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT Sets the maximum stack count of any ItemStack with an Item using this Settings instance.
+			COMMENT
+			COMMENT <p>Note that a count over 64 leads to unreliable behavior in GUIs.
+			COMMENT Damageable items can not have a maximum stack count (they default to 1).
+			COMMENT An Item.Settings' max count defaults to 64.
+			COMMENT
+			COMMENT @param maxCount maximum stack count of any  ItemStack with an item using this Settings instance
+			COMMENT @throws RuntimeException if this Settings instance also has a max damage value set
+			COMMENT @return this instance
 			ARG 1 maxCount
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:252]:
```suggestion
			ARG 1 maxCount
				COMMENT maximum stack count of any  ItemStack with an item using this Settings instance
```

File: mappings/net/minecraft/item/Item.mapping [hunk 14]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT @return this instance
 		METHOD method_7889 maxCount (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the maximum stack count of any ItemStack with an Item using this Settings instance.
+			COMMENT
+			COMMENT <p>Note that a count over 64 leads to unreliable behavior in GUIs.
+			COMMENT Damageable items can not have a maximum stack count (they default to 1).
+			COMMENT An Item.Settings' max count defaults to 64.
+			COMMENT
+			COMMENT @param maxCount maximum stack count of any  ItemStack with an item using this Settings instance
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 15]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT @throws RuntimeException if this Settings instance also has a max damage value set
+			COMMENT @return this instance
 			ARG 1 maxCount
 		METHOD method_7892 group (Lnet/minecraft/class_1761;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the ItemGroup of any item using this Settings instance. ItemGroups represent tabs in the creative inventory.
+			COMMENT
+			COMMENT @param group ItemGroup to use
+			COMMENT @return this instance
 			ARG 1 group
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:258]:
```suggestion
			ARG 1 group
				COMMENT the item group to use
```

File: mappings/net/minecraft/item/Item.mapping [hunk 16]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT An Item.Settings' max count defaults to 64.
+			COMMENT
+			COMMENT @param maxCount maximum stack count of any  ItemStack with an item using this Settings instance
+			COMMENT @throws RuntimeException if this Settings instance also has a max damage value set
+			COMMENT @return this instance
 			ARG 1 maxCount
 		METHOD method_7892 group (Lnet/minecraft/class_1761;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the ItemGroup of any item using this Settings instance. ItemGroups represent tabs in the creative inventory.
+			COMMENT
+			COMMENT @param group ItemGroup to use
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 17]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT @return this instance
 			ARG 1 group
 		METHOD method_7894 rarity (Lnet/minecraft/class_1814;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the {@link Rarity} of any item configured with this Settings instance, which changes the color of its name.
+			COMMENT
+			COMMENT <p>An item's rarity defaults to {@link Rarity.COMMON}.
+			COMMENT
+			COMMENT @param rarity rarity to apply to items using this Settings instance
+			COMMENT @return this instance
 			ARG 1 rarity
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:266]:
```suggestion
			ARG 1 rarity
				COMMENT rarity to apply to items using this Settings instance
```

File: mappings/net/minecraft/item/Item.mapping [hunk 18]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT @param group ItemGroup to use
+			COMMENT @return this instance
 			ARG 1 group
 		METHOD method_7894 rarity (Lnet/minecraft/class_1814;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the {@link Rarity} of any item configured with this Settings instance, which changes the color of its name.
+			COMMENT
+			COMMENT <p>An item's rarity defaults to {@link Rarity.COMMON}.
+			COMMENT
+			COMMENT @param rarity rarity to apply to items using this Settings instance
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 19]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT @return this instance
 			ARG 1 rarity
 		METHOD method_7895 maxDamage (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the maximum durability of any item configured with this Settings instance.
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when this method is called.
+			COMMENT
+			COMMENT @param maxDamage maximum durability of an ItemStack using an item with this Item.Settings instance
+			COMMENT @return this instance
 			ARG 1 maxDamage
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:274]:
```suggestion
			ARG 1 maxDamage
				COMMENT maximum durability of an ItemStack using an item with this Item.Settings instance
```

File: mappings/net/minecraft/item/Item.mapping [hunk 20]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT @param rarity rarity to apply to items using this Settings instance
+			COMMENT @return this instance
 			ARG 1 rarity
 		METHOD method_7895 maxDamage (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the maximum durability of any item configured with this Settings instance.
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when this method is called.
+			COMMENT
+			COMMENT @param maxDamage maximum durability of an ItemStack using an item with this Item.Settings instance
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 21]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT @return this instance
 			ARG 1 maxDamage
 		METHOD method_7896 recipeRemainder (Lnet/minecraft/class_1792;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the recipe remainder for any item configured with this Settings instance. When an item with a recipe remainder is used in a crafting recipe, the remainder is left in the table or returned to the player.
+			COMMENT
+			COMMENT @param recipeRemainder remainder item to return when an item with this Settings instance is used in a crafting recipe
+			COMMENT @return this instance
 			ARG 1 recipeRemainder
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:281]:
```suggestion
			ARG 1 recipeRemainder
				COMMENT remainder item to return when an item with this Settings instance is used in a crafting recipe
```

File: mappings/net/minecraft/item/Item.mapping [hunk 22]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when this method is called.
+			COMMENT
+			COMMENT @param maxDamage maximum durability of an ItemStack using an item with this Item.Settings instance
+			COMMENT @return this instance
 			ARG 1 maxDamage
 		METHOD method_7896 recipeRemainder (Lnet/minecraft/class_1792;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the recipe remainder for any item configured with this Settings instance. When an item with a recipe remainder is used in a crafting recipe, the remainder is left in the table or returned to the player.
+			COMMENT
+			COMMENT @param recipeRemainder remainder item to return when an item with this Settings instance is used in a crafting recipe
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 23]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT @return this instance
 			ARG 1 recipeRemainder
 		METHOD method_7898 maxDamageIfAbsent (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Calls {@link Item.Settings#maxDamage} If this Settings instance has not already set max damage (or if max damage is the default value, 0).
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when maxDamage is called.
+			COMMENT
+			COMMENT @param maxDamage maximum durability of an ItemStack using an item with this Item.Settings instance
+			COMMENT @return this instance
 			ARG 1 maxDamage
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping:288]:
```suggestion
			ARG 1 maxDamage
				COMMENT maximum durability of an ItemStack using an item with this Item.Settings instance
```

File: mappings/net/minecraft/item/Item.mapping [hunk 24]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT @param recipeRemainder remainder item to return when an item with this Settings instance is used in a crafting recipe
+			COMMENT @return this instance
 			ARG 1 recipeRemainder
 		METHOD method_7898 maxDamageIfAbsent (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Calls {@link Item.Settings#maxDamage} If this Settings instance has not already set max damage (or if max damage is the default value, 0).
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when maxDamage is called.
+			COMMENT
+			COMMENT @param maxDamage maximum durability of an ItemStack using an item with this Item.Settings instance
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
```

File: mappings/net/minecraft/item/Item.mapping [hunk 25]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT
+			COMMENT <p>An item's rarity defaults to {@link Rarity.COMMON}.
+			COMMENT
+			COMMENT @param rarity rarity to apply to items using this Settings instance
+			COMMENT @return this instance
 			ARG 1 rarity
 		METHOD method_7895 maxDamage (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the maximum durability of any item configured with this Settings instance.
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when this method is called.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
			COMMENT <p>Note that max stack count is set to 1 when this method is called.
```

File: mappings/net/minecraft/item/Item.mapping [hunk 26]
```diff
@@ -213,17 +230,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
 		METHOD method_7896 recipeRemainder (Lnet/minecraft/class_1792;)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Sets the recipe remainder for any item configured with this Settings instance. When an item with a recipe remainder is used in a crafting recipe, the remainder is left in the table or returned to the player.
+			COMMENT
+			COMMENT @param recipeRemainder remainder item to return when an item with this Settings instance is used in a crafting recipe
+			COMMENT @return this instance
 			ARG 1 recipeRemainder
 		METHOD method_7898 maxDamageIfAbsent (I)Lnet/minecraft/class_1792$class_1793;
+			COMMENT Calls {@link Item.Settings#maxDamage} If this Settings instance has not already set max damage (or if max damage is the default value, 0).
+			COMMENT
+			COMMENT <p>Note that max stack count is set to 0 when maxDamage is called.
```
Comment (liach) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
			COMMENT <p>Note that max stack count is set to 1 when maxDamage is called.
```

**Review by liach — CHANGES_REQUESTED** [review]
Some of these won't pass javadoc linting. This should be a comprehensive list of suggestions after which the javadoc would be good (hope I didn't miss anything!)

**Comment by liach** [comment]
@Draylar Mind apply those suggestions real quick? This should be read to go once all my suggestions are applied.

**Comment by Draylar** [comment]
> @Draylar Mind apply those suggestions real quick? This should be read to go once all my suggestions are applied.

Thanks for the reminder! 

\* Also, I am only now realizing there is a batch merge-suggestions button. Sorry for the commit spam.

**Comment by Technici4n** [comment]
@Draylar note that the build is failing.

File: mappings/net/minecraft/item/Item.mapping [hunk 27]
```diff
@@ -18,11 +18,14 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
 		COMMENT Checks if this item is food and therefore is edible.
 	METHOD method_19264 getFoodComponent ()Lnet/minecraft/class_4174;
+		COMMENT Returns this item's {@link FoodComponent}, or {@code null} if one was not set.
 	METHOD method_21830 getEatSound ()Lnet/minecraft/class_3414;
 	METHOD method_21831 getDrinkSound ()Lnet/minecraft/class_3414;
 	METHOD method_24357 damage (Lnet/minecraft/class_1282;)Z
+		COMMENT Returns whether this item can be damaged by the given {@link DamageSource source}.
 		ARG 1 source
 	METHOD method_24358 isFireproof ()Z
+			COMMENT Returns whether this item is immune to fire and lava damage.
```
Comment (enbrain) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
	METHOD method_24358 isFireproof ()Z
		COMMENT Returns whether this item is immune to fire and lava damage.
```

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/item/Item.mapping (hunk 13 above)
Comment (enbrain) [reply mappings/net/minecraft/item/Item.mapping:252]:
```suggestion
			ARG 1 maxCount
				COMMENT maximum stack count of any ItemStack with an item using this Settings instance
```
Fixed typo (unnecessary space between "any" and "ItemStack").

**Comment by enbrain** [comment]
Also note that some suggestions from liach are hidden.

**Comment by liach** [comment]
@Draylar no worries, those failures are from javadoc lint; otherwise it doesn't report any problem. 

A few suggestions from https://github.com/FabricMC/yarn/pull/1792#discussion_r614885686 should be taken.

On a side note, to batch apply suggestions, you must go into the files changed tab of a pr first. This is unintuitive but it's the design.

File: mappings/net/minecraft/item/Item.mapping (hunk 4 above)
Comment (haykam821) [reply mappings/net/minecraft/item/Item.mapping]: Do note that Yarn branches for newer versions of Minecraft are built using Java 16, so inline return tags can be used. This pull request should be updated accordingly.

For more information, see #2454.

**Comment by modmuss50** [comment]
What needs to be done here? Seems quite close to being ready to merge.

**Comment by haykam821** [comment]
This pull request still has some style issues like documenting parameters through `@param` and not having links.

**Comment by liach** [comment]
Yeah. Unfortunately draylar doesn't allow maintainer to push changes, so I guess some triage guy will pick it up? In addition, with the migration to java 16, we are using `{@return}` inline tags, and this pr needs such an update as well.

**Comment by IotaBread** [comment]
Opened a PR to this branch addressing the issues
https://github.com/Draylar/yarn/pull/1

**Review by IotaBread — APPROVED** [review]

**Review by liach — APPROVED** [review]
:rocket:

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/item/Item.mapping [hunk 28]
```diff
@@ -213,17 +231,59 @@ CLASS net/minecraft/class_1792 net/minecraft/item/Item
...
+			COMMENT Sets the maximum stack count of any ItemStack with an Item using this Settings instance.
+			COMMENT
+			COMMENT <p>Note that a count over 64 leads to unreliable behavior in GUIs.
+			COMMENT Damageable items can not have a maximum stack count (they default to 1).
+			COMMENT An Item.Settings' max count defaults to 64.
+			COMMENT
+			COMMENT @throws RuntimeException if this Settings instance also has a max damage value set
+			COMMENT @return this instance
 			ARG 1 maxCount
+				COMMENT maximum stack count of any  ItemStack with an item using this Settings instance
```
Comment (enbrain) [inline mappings/net/minecraft/item/Item.mapping]:
```suggestion
				COMMENT maximum stack count of any ItemStack with an item using this Settings instance
```

**Review by enbrain — APPROVED** [review]

File: mappings/net/minecraft/item/Item.mapping (hunk 28 above)
Comment (liach) [reply mappings/net/minecraft/item/Item.mapping]: Hmm, @Draylar mind consider taking this real quick? We can probably open an issue for assorted javadoc typos in the future.

**Pull request merged**